## RNA-SEQ PIPELINE

A quick pipeline with minimum settings, use with discretion.

This pipeline performs the following tasks:
- perform quality control check on FastQ files (using FastQC and Trimmomatic)
- align reads of each sample in a run against reference genome (using Hisat2)
- parsing the alignment file (using Samtools and Sambamba)
- count reads and calculate normalized value in features (using Stringtie)
- perform DE analysis for standard designs (using DESeq2)
- enrichment analysis (using clusterProfiler)


## Installation
Download the RNA-seq pipeline.

```bash
git clone https://github.com/hr1912/RNA-seq.git
```

Make sure all [dependencies](#dependencies) are installed and the right paths are set.

## Usage
#### Run pipeline

For each sample, run quality control and alignment:

```bash
cd <work directory>
sh 01.trim_map.sh <sample_name> <rd1.raw.fastq.gz> <rd2.raw.fastq.gz>
```
and reads counting and normalization:
```bash
cd <sample_name>
sh 02.stringtie.sh <sample_name>
```

Then merge all the "table.txt" file generated by reads counting using:

```bash
/usr/bin/python2.7 prepDE.py -i file.lst
```

"file.lst" is a text file with each row looks like

'<sample_name>  <sample.table.txt>'

Then a "gene_count_matrix.csv" is generated.

Refer to "analysis. R" for differential expression analysis and enrichment analysis. 

## Dependencies

#### Core tools
- Python 2.7
- R 3.5.1 or higher
- Java 1.7 or higher

#### Bio tools
- [FastQC](http://www.bioinformatics.babraham.ac.uk/projects/fastqc/)
- [Trimmomatic](http://www.usadellab.org/cms/?page=trimmomatic)
- [Hisat2](http://daehwankimlab.github.io/hisat2/)
- [Sambamba](http://lomereiter.github.io/sambamba/)
- [Samtools](https://github.com/CuppenResearch/bamMetrics)
- [Stringtie](https://ccb.jhu.edu/software/stringtie/)

#### R packages
- DESeq2
- Tidyverse suite
- RColorBrewer
- clusterProfile
- pheatmap
- Bioconductor annotations for your reference:
  - Human: org.Hs.eg.db

## Generate genome indexes
Create a directory where you want to store the indexes (e.g. /genome_reference/Hisat2/Homo_sapiens.GRCh38).

Collect the following files for your genome:
- Fasta file containing the genome reference sequences
- GTF file containing annotated transcripts

Run Hisat2:
```bash
hisat2-build
	-p 8
	--ss /path/Homo_sapiens.GRCh38.splicesites.txt \
	/path/Homo_sapiens.GRCh38 \
	/path/genome_reference/Hisat2/Homo_sapiens.GRCh38
```
